core/core_builders.py

```diff
@@ -47,27 +47,68 @@ def version_hash_builder(target, source, env):
 
 
 def encryption_key_builder(target, source, env):
-    src = source[0].read() or "0" * 64
+    src_key = source[0].read() or "0" * 64
+    xor_mask = source[1].read() or "0" * 2
+
+    try:
+        if len(xor_mask) != 2:
+            raise ValueError
+        else:
+            xor_mask = int(xor_mask)
```

Comment:
Specify base 16 here(`xor_mask = int(xor_mask, 16)`)? You must specify 0 as the base in order to invoke this prefix-guessing behavior; if you omit the second parameter, [int()](https://docs.python.org/library/functions.html#int) will assume base-10. And since there's a length check before this, this will never work in its current state.